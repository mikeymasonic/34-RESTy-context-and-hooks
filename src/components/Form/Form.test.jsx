import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form 
      url=""
      method=""
      body=""
      disable={true}
      onChange={() => {}}
      onSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
