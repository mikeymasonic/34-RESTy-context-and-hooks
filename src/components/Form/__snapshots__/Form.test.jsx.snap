// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<div
  className="Form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      URL:
      <input
        name="url"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <div>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="GET"
        />
        GET
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="POST"
        />
        POST
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="PUT"
        />
        PUT
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="PATCH"
        />
        PATCH
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="DELETE"
        />
        DELETE
      </label>
    </div>
    <textarea
      disabled={true}
      name="body"
      onChange={[Function]}
      placeholder="JSON Body"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
